tsinfer/inference.py

```diff
@@ -1076,6 +1099,42 @@ def __init__(
             for _ in range(num_threads)
         ]
 
+    @staticmethod
+    def recombination_rate_to_dist(rho, positions):
+        """
+        Return the mean number of recombinations between adjacent positions (i.e.
+        the genetic distance in Morgans) given either a fixed rate or a RateMap
+        """
+        try:
+            return np.diff(rho.get_cumulative_mass(positions))
+        except AttributeError:
+            return np.diff(positions) * rho
+
+    @staticmethod
+    def recombination_dist_to_prob(genetic_distances):
```

Comment:
These functions are great. I think we just need a handful of test cases that illustrate the behaviour on small simple examples where we pre-calculate the expected result. This will make it easier to see what they do, also.

Reply:
I've just pushed a few more commits with a load of tests for these functions. Look in particular for `test_inference.py`: `test_mismatch_ratio_to_prob_known`. Is that the sort of thing you mean @jeromekelleher ?

I put these as new commits so you can see the changes easily, but once you've OKed it I'll squash them all together, I guess?